pkgs/by-name/ho/hof/package.nix

```
@@ -0,0 +1,35 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+let
+  pname = "hof";
+  version = "0.6.9-beta.1";
+in
+buildGoModule {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner  = "hofstadter-io";
+    repo   = "hof";
+    rev    = "v${version}";
+    hash   = "sha256-4yVP6DRHrsp52VxBhr7qppPhInYEsvPbIfxxQcRwHTw=";
+  };
```

```suggestion

buildGoModule rec {
  pname = "hof";
  version = "0.6.9-beta.1";

  src = fetchFromGitHub {
    owner  = "hofstadter-io";
    repo   = "hof";
    rev    = "v${version}";
    hash   = "sha256-4yVP6DRHrsp52VxBhr7qppPhInYEsvPbIfxxQcRwHTw=";
  };
```

you can use rec to make it recursive so you can reuse the attributes in the same attrSet. We usually don't use the let binding for pname or version.

---

@Janik-Haag  Thank you for taking a look at this.
I was under impression that let is preferred over rec . Sorry to put @happysalada on the spot. https://github.com/NixOS/nixpkgs/pull/227858/

I personally don't mind let. Nor rec. I just hope we make it easier to understand for newcomers what is acceptable approach.

---

as far as I know rec is preferred in this case but I also don't care too much to be honest.

---

I usually have no preference between `let..in` vs `rec` and the expression is not too complex where `rec` would be hard to follow. I just subscribed to the "best practices" outlined [here](https://nix.dev/guides/best-practices#recursive-attribute-set-rec) but I'll defer to your suggestion 😀

---

Haha, no worries. I personally prefer not using rec, but its definitely a matter of preference.

I would say if you feel strongly about it either way feel free to argue it, i think we are open in the community.

If you dont have any strong feelings about it, feel free to go with the flow, your PR will get merged faster :-)

---

Just to try to bring to the debate.

Rec generates a bit of indirection on which variables is being used in different places, but it has the advantage of being more terse.
I think its a close call and really depends on what you prefer, terse vs obvious. (Obvious is also relative i think, its more obvious for new comers, but people used to it dont care that much anymore).

(The guy writing the guide you linked also thinks obvious beats terse)